libs/prelude/Prelude/Interfaces.idr

```diff
@@ -372,6 +372,145 @@ namespace Foldable
     foldl = foldl . foldl
     null tf = null tf || all (force . null) tf
 
+||| `Bifoldable` identifies foldable structures with two different varieties
+||| of elements (as opposed to `Foldable`, which has one variety of element).
+||| Common examples are `Either` and `Pair`.
+public export
+interface Bifoldable p where
+  bifoldr : (a -> acc -> acc) -> (b -> acc -> acc) -> acc -> p a b -> acc
+
+  bifoldl : (acc -> a -> acc) -> (acc -> b -> acc) -> acc -> p a b -> acc
+  bifoldl f g z t = bifoldr (flip (.) . flip f) (flip (.) . flip g) id t z
+
+  binull : p a a -> Lazy Bool
```

Comment:
Why are we suddenly taking `p a a` instead of `p a b`?

Reply:
Thanks for catching this. Fixed.